pkg/service/backup/restore_worker_data.go

```diff
@@ -247,7 +270,9 @@ func (w *restoreWorker) workFunc(ctx context.Context, run *RestoreRun, target Re
 			pr.setRestoreCompletedAt()
 			w.insertRunProgress(ctx, pr)
 
-			w.metrics.UpdateRestoreProgress(w.ClusterID, pr.ManifestPath, pr.Keyspace, pr.Table, pr.Downloaded+pr.Skipped)
+			restoredBytes := pr.Downloaded + pr.Skipped
+			w.metrics.UpdateRestoreProgress(w.ClusterID, pr.ManifestPath, pr.Keyspace, pr.Table, restoredBytes)
+			w.metrics.DecreaseRemainingBytes(w.ClusterID, w.location, w.SnapshotTag, target.Keyspace, restoredBytes)
```

Comment:
What about when batch returns to the pool when given node encounters an error?
I guess the metrics should be increased in this case.

Reply:
I don't really follow.
Remaining bytes are decreased when we see that load&stream ended up with the success.
https://github.com/scylladb/scylla-manager/blob/01994811ae1c6736bafc4edf203106b095a93659/pkg/service/backup/restore_worker_data.go#L283-L300

Is there something happening in parallel that I'm not aware of ?

Reply:
It looks like I attached wrong lines of code, my bad.
I was concerned about the `batch_size` metric and its behavior, when some node (after assembling the batch) encountered an error and returned its batch to the pool (so that the other nodes might try to restore it).
Should we decrease the `batch_size` metric in this case?

Reply:
I still don't follow.
Metric is not decreased if batch is returned to the pool.
It decreases it only if there is no error on downloading and no error on load&streaming when node tries to restore the batch.

Reply:
```go
	batch := w.batchFromIDs(pr.SSTableID)
	w.updateBatchSizeMetric(ctx, pr.ClusterID, batch, pr.Host, srcDir)

	w.cleanUploadDir(ctx, pr.Host, dstDir, batch)

	jobID, err := w.startDownload(ctx, pr.Host, dstDir, srcDir, batch)
	if err != nil {
		w.deleteRunProgress(ctx, pr)
		w.returnBatchToPool(pr.SSTableID)
		return err
	}
	pr.AgentJobID = jobID
	w.insertRunProgress(ctx, pr)
	return nil
}
```

This seems to be the only place where the `batch_size` metric is increased. My question is, shouldn't we decrease it inside `w.returnBatchToPool(pr.SSTableID)` function? Or is it fine that the sum over `batch_size` might be greater than the actual size of all batches (in case of error)?

Reply:
now I've got it,
Yes, batch_size should be decreased.

Thanks !